toolchain/parser/parser.cpp

```diff
@@ -1662,6 +1782,7 @@ auto Parser::HandleVarFinishAsForState() -> void {
 
 auto Parser::HandleInterfaceIntroducerState() -> void {
   auto state = PopState();
+  stack_context_ = ParseContext::Interface;
```

Comment:
```suggestion
  CARBON_CHECK(stack_context_ == ParseContext::File) << "TODO: Support nesting.";
  stack_context_ = ParseContext::Interface;
```

Let's add a CHECK to make it clear this isn't fully implemented.